src/mainwindow.cpp

```diff
@@ -1000,21 +1003,19 @@ void MainWindow::on_centralWidget_customContextMenuRequested(const QPoint &pos)
   createPopupMenu()->exec(ui->centralWidget->mapToGlobal(pos));
 }
 
-void MainWindow::scheduleUpdateButton()
+void MainWindow::scheduleCheckForProblems()
 {
   if (!m_UpdateProblemsTimer.isActive()) {
-    m_UpdateProblemsTimer.start(1000);
+    m_UpdateProblemsTimer.start(500);
```

Comment:
I'm still not fan of that 500ms here, but I understand your point. Maybe that could be a setting (only in the `.ini`, much like the number of threads)?

Reply:
I don't see changing this value from the ini as particularly useful and I think we did already enough for this 1 second off of startup time on a worst case scenario (I tested again it was doing the ~500ms three times).

Reply:
I don't understand the end of your comment. It was running the `checkForProblems` 3 times? Or did it update the timer 3 times? If it updates the timer 3 times, what matters is more the time between the update rather than the actual number of calls (e.g., if you update the timer 3 times with 150ms between each, then 200ms would be sufficient).

Reply:
Originally for with no changes I was getting three updateProblemsButton() that took ~500ms and another 2 or 3 that took 0ms on startup. Now I only get one 450ms refresh that is actually performed asynchronously, so I guess we are saving an incredible 1.5 seconds in worst case scenario hahaha

Reply:
Yes, but since this has no impact on startup anymore, I feel like having a shorter time-out here would make some part of the UI more responsive.

Reply:
Well it's just 500 ms that affects exclusively the problems button icon so I think it's worth doing a single refresh instead of having it refreshed 2 or 3 times because we had a delay that was too short.